inference-engine/src/transformations/src/transformations/common_optimizations/pull_transpose_through_fq.cpp

```diff
@@ -9,56 +9,42 @@
 
 #include <ngraph/opsets/opset1.hpp>
 #include <ngraph/rt_info.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
 
 NGRAPH_RTTI_DEFINITION(ngraph::pass::PullTransposeThroughFQUp, "PullTransposeThroughFQUp", 0);
 
 ngraph::pass::PullTransposeThroughFQUp::PullTransposeThroughFQUp() {
-    auto data1 = std::make_shared<pattern::op::Label>(element::f32, Shape{1, 1, 1, 1});
-    auto data2 = std::make_shared<pattern::op::Label>(element::f32, Shape{1, 1, 1, 1});
-    auto data3 = std::make_shared<pattern::op::Label>(element::f32, Shape{1, 1, 1, 1});
-    auto data4 = std::make_shared<pattern::op::Label>(element::f32, Shape{1, 1, 1, 1});
-    auto data5 = std::make_shared<pattern::op::Label>(element::f32, Shape{1, 1, 1, 1});
-    auto fq = std::make_shared<ngraph::opset1::FakeQuantize>(data1, data2, data3, data4, data5, 1);
-    auto transpose_order = std::make_shared<pattern::op::Label>(element::i64, Shape{4});
-    auto transpose = std::make_shared<ngraph::opset1::Transpose>(fq, transpose_order);
+    auto m_fq = pattern::wrap_type<opset1::FakeQuantize>({pattern::any_input(pattern::has_static_rank()),
+                                                          pattern::any_input(pattern::has_static_rank()),
+                                                          pattern::any_input(pattern::has_static_rank()),
+                                                          pattern::any_input(pattern::has_static_rank()),
+                                                          pattern::any_input(pattern::has_static_rank())},
+                                                          pattern::consumers_count(1));
+    auto m_transpose = pattern::wrap_type<opset1::Transpose>({m_fq, pattern::wrap_type<opset1::Constant>()});
 
-    ngraph::graph_rewrite_callback callback = [](pattern::Matcher& m) {
-        auto transpose = ngraph::as_type_ptr<ngraph::opset1::Transpose>(m.get_match_root());
-        if (!transpose) {
-            return false;
-        }
-
-        auto const_node = transpose->input(1).get_source_output().get_node_shared_ptr();
-        auto const_order = ngraph::as_type_ptr<ngraph::opset1::Constant>(const_node);
-        if (!const_order) {
-            return false;
-        }
-
-        auto fq_node = transpose->input(0).get_source_output().get_node_shared_ptr();
-        auto fq = ngraph::as_type_ptr<ngraph::opset1::FakeQuantize>(fq_node);
-        if (!fq || fq->output(0).get_target_inputs().size() != 1) {
-            return false;
-        }
+    ngraph::matcher_pass_callback callback = [=](pattern::Matcher& m) {
+        auto & pattern_map = m.get_pattern_value_map();
+        auto transpose = pattern_map[m_transpose].get_node_shared_ptr();
+        auto fq = pattern_map[m_fq].get_node_shared_ptr();
 
-        auto input_shape = fq->input(0).get_source_output().get_shape();
+        auto input_shape = fq->input(0).get_partial_shape().rank().get_length();
```

Comment:
nit: it is not input shape, from my point of view input_rank is a better name.

Reply:
fixed.